### PR TITLE
Simple event dispatcher fix

### PR DESCRIPTION
**Pull request by Manriel**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Doc update?   | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | 
| License       | MIT

Preventing the flushing of the bubble queue when event handler raises another event

Unit test case and the fix.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/.idea
```
Comment (fabpot) [inline .gitignore]: To be reverted

Comment (Manriel) [reply .gitignore]: Done

**Comment by Manriel** [comment]
@fabpot any update on this?

**Comment by fabpot** [comment]
Thank you @Manriel.

**Pull request merged**